tests/functional/s/super/superfluous_parens.py

```diff
@@ -18,10 +19,10 @@
 del(DICT['b'])  # [superfluous-parens]
 del DICT['a']
 
-B = [x for x in ((3, 4))]  # [superfluous-parens]
+B = [x for x in ((3, 4))]
 C = [x for x in ((3, 4) if 1 > 0 else (5, 6))]
-D = [x for x in ((3, 4) if 1 > 0 else ((5, 6)))]  # [superfluous-parens]
-E = [x for x in ((3, 4) if 1 > 0 else ((((5, 6)))))]  # [superfluous-parens]
+D = [x for x in ((3, 4) if 1 > 0 else ((5, 6)))]
```

Comment:
Why are the parens around `(5, 6)` not considered superfluous here?

Reply:
See the report in https://github.com/PyCQA/pylint/issues/4907

There is no good way to determine whether the parenthesises are there by accident or if they are intentional.

Reply:
Apologies if I'm missing something obvious, but I think it should be possible if we re-implement the checker to consume the AST instead of tokens. At the AST level we would immediately know that `()` is an empty tuple and thus does require parentheses.

Reply:
We already consume the AST so that's not really the issue here. It might be possible to fix this, but in m'n initial attempt I created a lot of false positives so we reverted to not doing this kind of check on nested tuples in comprehension.

I welcome any PR that tries to fix this though!